core/src/banking_stage.rs

```diff
@@ -1915,6 +1972,31 @@ impl BankingStage {
             "packet_conversion",
         );
 
+        // Filter out extraneous consensus transactions
+        let (filtered_extraneous_votes_indexes, _filter_extraneous_votes_time) = measure!({
+            if bank
+                .feature_set
+                .is_active(&allow_votes_to_directly_update_vote_state::id())
+            {
+                let indexes =
+                    Self::filter_extraneous_votes(&transactions, &transaction_to_packet_indexes);
```

Comment:
Filtering votes here within a batch works, functionally, but agree with @carllin that ideally the filtering should be done once on the packets in buffer. Perhaps can add additional plumbing to mark deserialized packet in `unprocessed_packet_batches` as `discard` if its sanitized transaction is identified as extraneous vote.